### ISSUE TITLE
Guarantee `Interface.implements` returns a `Bool`

### ISSUE THREAD

**Issue by nickrobinson251** [quinnj/Interfaces.jl#23]
I need to make a MWE of this, but i think our current code can result in returning `nothing`

i think it's in the case where an `@interface` definiton ends in a `if .. elseif ... end` block

**Comment by quinnj** [comment]
Yeah, I can see this is _possible_, because we're currently relying on the interface author to ensure things handled correctly. For example, in this:
```julia
Base.IndexStyle(::Type{T})::Union{Base.IndexLinear, Base.IndexCartesian}

    if Base.IndexStyle(T) == Base.IndexLinear()
        getindex(::T, i::Int)
        setindex!(::T, v, i::Int)
    elseif Base.IndexStyle(T) == Base.IndexCartesian()
        getindex(::T, I::Vararg{Int})
        setindex!(::T, v, I::Vararg{Int})
    end
```
We first ensure that `IndexStyle` returns one of two options, and then our `if-elseif` checks both those options. Similarly for the `AbstractTable` case, we first check `isrowtable` and `columnaccess`, then do an if-elseif block.

So it would be bad if the author didn't first check definitions, but just went straight into an if-elseif block. We could perhaps insert an error to be thrown in a generated `else` block in that case?
